### PR TITLE
Add server to discord bridge

### PR DESCRIPTION
**Pull request by ChillerDragon**
ddnet-discord.py does all the magic. It takes an string as argument and sends it too an discord webhook.
The keys.py could be hidden by gitignore for example, but for now it holds my webhook super secret key so you can test it on my server if you want.

To test it the server needs ``ddnet-discord.py`` and ``keys.py`` in the same directory as the server.
Then an ingame command like ``/modhelp help me please c:`` should arrive at the https://discord.gg/HsTAyzp discord.

For security i did replace some chars like ` and ' with spaces to avoid some escapes, but im not sure if it is really secure. I guess its dangerous since we have userinput in a system call o.O

### CODE REVIEW

File: scripts/ddnet-discord.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+# Python webhook to discord
+
+import requests
+import keys
+import sys
+
+def send_discord(message):
+    requests.post("https://discordapp.com/api/webhooks/" + keys.webhook, data={"content": message})
+
+if not len(sys.argv) == 2:
```
Comment (def-) [inline scripts/ddnet-discord.py]: `!=`

File: scripts/ddnet-discord.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    print('usage: ' + str(sys.argv[0]) + ' "message"')
```
Comment (def-) [inline scripts/ddnet-discord.py]: `message`, not `"message"`

File: src/game/server/ddracechat.cpp [hunk 3]
```diff
@@ -1438,6 +1438,39 @@ void CGameContext::ConModHelp(IConsole::IResult *pResult, void *pUserData)
 			pSelf->Server()->ClientName(pResult->m_ClientID),
 			pResult->m_ClientID);
 
+    // Prepare user data for discord
+    char aHelpMsg[128];
+    char aName[32];
+    str_copy(aName, pSelf->Server()->ClientName(pResult->m_ClientID), sizeof(aName));
+    str_copy(aHelpMsg, pResult->GetString(0), sizeof(aHelpMsg));
+    for(unsigned int i = 0; i < sizeof(aHelpMsg); i++) // don't escape the discord format
```
Comment (def-) [inline src/game/server/ddracechat.cpp]: Why is it escaped?

Comment (ChillerDragon) [reply src/game/server/ddracechat.cpp]: I use the `` markdown in the message http://puu.sh/AumHY/506085a1df.png
And the python script takes one string thats why i wrote "message" to take multiple words as one argument. And if the name or the help message includes a ' sombody could execute a systemcall.

File: scripts/ddnet-discord.py (hunk 1 above)
Comment (def-) [reply scripts/ddnet-discord.py]: Generally you try to use the shorter construct if it has the same ability and clarity. `if len(sys.argv) != 2` is faster and easier to read than `if not len(sys.argv) == 2`.

**Comment by def-** [comment]
So who wants to rewrite it to use libcurl? ;) @Learath2  @heinrich5991

**Comment by 12pm** [comment]
Does it prevent abuse somehow? If people could spam it all the time for no actual reason, it would have the opposite effect of what it's trying to achieve

**Comment by ChillerDragon** [comment]
The command has a time delay already. You can test it on ddnet servers already. @12pm

**Comment by Learath2** [comment]
@heinrich5991 you ported this over to libcurl right?

**Comment by heinrich5991** [comment]
I've worked on this at least. :)

**Comment by heinrich5991** [comment]
@daverck: See #1157.

Superseded by #1157. Thanks @ChillerDragon.

**Pull request closed without merging**